rust/lance/src/index/vector/hnsw.rs

```diff
@@ -0,0 +1,267 @@
+            // ROW_ID_FIELD.clone(),
+            arrow_schema::Field::new("_node_id", DataType::UInt32, true),
+        ]));
+        Ok(RecordBatch::try_new(
+            schema,
+            vec![distances, Arc::new(node_ids)],
+        )?)
+    }
+
+    fn is_loadable(&self) -> bool {
+        true
+    }
+
+    fn use_residual(&self) -> bool {
+        false
+    }
+
+    fn check_can_remap(&self) -> Result<()> {
+        Ok(())
+    }
+
+    async fn load(
+        &self,
+        reader: Arc<dyn Reader>,
+        _offset: usize,
+        _length: usize,
+    ) -> Result<Box<dyn VectorIndex>> {
+        let schema = Schema::try_from(&arrow_schema::Schema::new(vec![
+            NEIGHBORS_FIELD.clone(),
+            VECTOR_ID_FIELD.clone(),
+        ]))?;
+
+        let reader = FileReader::try_new_from_reader(
+            reader.path(),
+            reader.clone(),
+            None,
+            schema,
+            0,
+            0,
+            None,
+        )
+        .await?;
+
+        let hnsw = HNSW::load(&reader, self.storage.clone().unwrap()).await?;
+
+        Ok(Box::new(Self {
+            hnsw,
+            storage: self.storage.clone(),
+            partition_metadata: self.partition_metadata.clone(),
+        }))
+    }
+
+    async fn load_partition(
+        &self,
+        reader: Arc<dyn Reader>,
+        offset: usize,
+        length: usize,
+        partition_id: usize,
+    ) -> Result<Box<dyn VectorIndex>> {
+        let schema = Schema::try_from(&arrow_schema::Schema::new(vec![
+            NEIGHBORS_FIELD.clone(),
+            VECTOR_ID_FIELD.clone(),
+        ]))?;
+
+        let reader = FileReader::try_new_from_reader(
+            reader.path(),
+            reader.clone(),
+            None,
+            schema,
+            0,
+            0,
+            None,
+        )
+        .await?;
+
+        let metadata = self.get_partition_metadata(partition_id)?;
+        let hnsw = HNSW::load_partition(
+            &reader,
+            offset..length,
+            self.metric_type(),
+            self.storage.clone().unwrap(),
+            metadata,
+        )
+        .await?;
+
+        Ok(Box::new(Self {
+            hnsw,
+            storage: self.storage.clone(),
+            partition_metadata: self.partition_metadata.clone(),
+        }))
+    }
+
+    fn remap(&mut self, _mapping: &HashMap<u64, Option<u64>>) -> Result<()> {
+        // This will be needed if we want to clean up IVF to allow more than just
+        // one layer (e.g. IVF -> IVF -> PQ).  We need to pass on the call to
+        // remap to the lower layers.
+
+        // Currently, remapping for IVF is implemented in remap_index_file which
+        // mirrors some of the other IVF routines like build_ivf_pq_index
+        Err(Error::Index {
+            message: "Remapping IVF in this way not supported".to_string(),
+            location: location!(),
+        })
```

Comment:
This comment can probably be removed, it's a little confusing.  Out of curiosity, does hnsw support remapping yet?

Reply:
not support, just remove this for now.